config/initializers/preload_stis.rb

```diff
@@ -0,0 +1,10 @@
+# https://guides.rubyonrails.org/autoloading_and_reloading_constants.html#option-4-preload-types-from-the-database
+
+if Rails.env.development? && Rails.application.config.eager_load
+  Rails.application.config.to_prepare do
+    ApplicationRecord.logger.silence do
+      types = User.unscoped.select(:type).distinct.pluck(:type)
+      types.compact.each(&:constantize)
+    end
+  end
+end
```

Comment:
I'm not sure why this is needed. I've read the linked docs, but I'm still not sure what the need is for preloading the different user classes.

When I check out the branch locally and run this in the Rails console:

```ruby
User.first.class.name
  User Load (0.3ms)  SELECT "users".* FROM "users" ORDER BY "users"."id" ASC LIMIT $1  [["LIMIT", 1]]
=> "Claims::SupportUser"
```

I get the expected result, and no weird log output, regardless of whether I have this initializer or not. It seems to work the same either way? 🤔 

Can you ELI5? 🙏🏻

Reply:
I messed up inverting the latter condition... It should be `if Rails.env.development? && !Rails.application.config.eager_load`

Anyway...

Try running `SupportUser.all` without this code. It will default to `where(type: "SupportUser")` in development as it isn't aware of the additional subclasses that inherit from `SupportUser`.